cmd/timestamp-cli/app/verify.go

```diff
@@ -85,11 +93,148 @@ func runVerify() (interface{}, error) {
 
 	return &verifyCmdOutput{TimestampPath: tsrPath}, err
 }
 
+func newVerifyOpts() (verification.VerifyOpts, error) {
+	opts := verification.VerifyOpts{}
+
+	oid, err := getOid()
+	if err != nil {
+		return verification.VerifyOpts{}, fmt.Errorf("failed to parse value from oid flag: %w", err)
+	}
+	if oid != nil {
+		opts.Oid = oid
+	}
+
+	certPathFlagVal := viper.GetString("cert")
+	if certPathFlagVal != "" {
+		cert, err := createCertFromPEMFile(certPathFlagVal)
+		if err != nil {
+			return verification.VerifyOpts{}, fmt.Errorf("failed to parse cert flag value from PEM file: %w", err)
+		}
+		opts.TsaCertificate = cert
+	}
+
+	roots, intermediates, err := getCerts()
+	if err != nil {
+		return verification.VerifyOpts{}, fmt.Errorf("failed to parse root and intermediate certs from cert-chain flag: %w", err)
+	}
+	if roots != nil && intermediates != nil {
+		opts.Roots = roots
+		opts.Intermediates = intermediates
+	}
+
+	nonce, err := getNonce()
+	if err != nil {
+		return verification.VerifyOpts{}, fmt.Errorf("failed to parse value from nonce flag: %w", err)
+	}
+	if nonce != nil {
+		opts.Nonce = nonce
+	}
+
+	subjectFlagVal := viper.GetString("subject")
+	if subjectFlagVal != "" {
+		opts.Subject = subjectFlagVal
+	}
+
+	return opts, nil
+}
+
+func getNonce() (*big.Int, error) {
+	nonceFlagVal := viper.GetString("nonce")
+	if nonceFlagVal == "" {
+		return nil, nil
+	}
+
+	nonce := new(big.Int)
+	nonce, ok := nonce.SetString(nonceFlagVal, 10)
+	if !ok {
+		return nil, fmt.Errorf("failed to convert string to big.Int")
+	}
+	return nonce, nil
+}
+
+func getCerts() ([]*x509.Certificate, []*x509.Certificate, error) {
+	certChainPEM := viper.GetString("cert-chain")
+	if certChainPEM == "" {
+		return nil, nil, nil
+	}
+
+	pemBytes, err := os.ReadFile(filepath.Clean(certChainPEM))
+	if err != nil {
+		return nil, nil, fmt.Errorf("error reading request from file: %w", err)
+	}
+	intermediateCerts := []*x509.Certificate{}
+	rootCerts := []*x509.Certificate{}
+	for len(pemBytes) > 0 {
+		block, rest := pem.Decode(pemBytes)
+		// if there is nothing left, we have found the last block
+		// which should be the root
+		cert, err := x509.ParseCertificate(block.Bytes)
+		if err != nil {
+			return nil, nil, fmt.Errorf("failed to parse certificate")
+		}
+		if rest == nil {
+			rootCerts = append(rootCerts, cert)
+		} else {
+			intermediateCerts = append(intermediateCerts, cert)
+		}
+		pemBytes = rest
+	}
+	return rootCerts, intermediateCerts, nil
+}
+
+func getOid() ([]int, error) {
+	oidFlagVal := viper.GetString("oid")
+	if oidFlagVal == "" {
+		return nil, nil
+	}
+
+	oidStrSlice := strings.Split(oidFlagVal, ".")
+	oid := make([]int, len(oidStrSlice))
+	for i, el := range oidStrSlice {
+		intVar, err := strconv.Atoi(el)
+		if err != nil {
+			return nil, err
+		}
+		oid[i] = intVar
+	}
+
+	return oid, nil
+}
+
+func createCertFromPEMFile(certPath string) (*x509.Certificate, error) {
```

Comment:
nit: I'd rename it to `parseTSACertificate(...)`, it looks like it just parses the certificate.